internal/config/authentication.go

```diff
@@ -529,3 +532,72 @@ func (a AuthenticationMethodGithubConfig) validate() error {
 
 	return nil
 }
+
+type AuthenticationMethodJWTConfig struct {
+	// ValidateClaims is used to validate the claims of the JWT token.
+	ValidateClaims struct {
+		// Issuer is the issuer of the JWT token.
+		Issuer string `json:"-" mapstructure:"issuer" yaml:"issuer,omitempty"`
+		// Audiences is the audience of the JWT token.
+		Audiences []string `json:"-" mapstructure:"audiences" yaml:"audiences,omitempty"`
+	} `json:"-" mapstructure:"validate_claims" yaml:"validate_claims,omitempty"`
+	// JWKsURL is the URL to the JWKS endpoint.
+	// This is used to fetch the public keys used to validate the JWT token.
+	JWKSURL string `json:"-" mapstructure:"jwks_url" yaml:"jwks_url,omitempty"`
+	// KeyFile is the path to the private PEM encoded key file on disk.
+	KeyFile string `json:"-" mapstructure:"key_file" yaml:"key_file,omitempty"`
```

Comment:
This should be the public key, no? Flipt server doesn't need the private key if we're supporting asymmetric key pairs. That is for clients to use when generating JWTs. If we supported something like `HS256` then we would need the private key.
```suggestion
	// PublicKeyFile is the path to the public PEM encoded key file on disk.
	PublicKeyFile string `json:"-" mapstructure:"public_key_file" yaml:"public_key_file,omitempty"`
```

Reply:
yeah it is public key. I agree renaming makes it clearer